app/src/main/java/ch/epfl/sweng/SDP/game/drawing/DrawingOffline.java

```diff
@@ -3,33 +3,68 @@
 import android.os.Bundle;
 import android.support.annotation.VisibleForTesting;
 import android.util.Log;
+import android.view.MotionEvent;
 import android.view.View;
 import android.widget.ImageView;
 import android.widget.RelativeLayout;
+import android.widget.TextView;
 
 import java.util.HashMap;
 import java.util.Random;
 
 import ch.epfl.sweng.SDP.R;
 import ch.epfl.sweng.SDP.game.drawing.items.Item;
+import ch.epfl.sweng.SDP.home.HomeActivity;
 import ch.epfl.sweng.SDP.localDatabase.LocalDbHandlerForImages;
+import ch.epfl.sweng.SDP.utils.LayoutUtils;
 
 /**
  * Class representing the practice mode, which is offline and where the user has the choice of
  * activating or not the mystery mode and the related item generation.
  */
 public class DrawingOffline extends GyroDrawingActivity {
 
+    private ImageView mysteryButton;
     private DrawingItems drawingItems;
     private boolean isToggled;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        drawingItems = new DrawingItems(this,
-                (RelativeLayout) findViewById(R.id.paintViewHolder), super.paintView,
-                new HashMap<Item, ImageView>(), new Random());
+        drawingItems = new DrawingItems(this, (RelativeLayout) findViewById(R.id.paintViewHolder),
+                super.paintView, new HashMap<Item, ImageView>(), new Random());
         isToggled = false;
+        mysteryButton = findViewById(R.id.mysteryModeButton);
+        setExitButtonListener();
+    }
+
+    private void setExitButtonListener() {
+        TextView exitButton = findViewById(R.id.exit);
+        final DrawingOffline activity = this;
+
+        exitButton.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View view, MotionEvent event) {
+                switch (event.getAction()) {
+                    case MotionEvent.ACTION_DOWN:
+                        LayoutUtils.pressButton(view, LayoutUtils.AnimMode.CENTER, activity);
+                        break;
+                    case MotionEvent.ACTION_UP:
```

Comment:
Consider refactoring this by adding this code in seperate functions. This code is way to long.